content/blog/2021/containerqueries.md

```diff
@@ -0,0 +1,146 @@
+
+In this demo (which only works in the latest version of [Chrome Canary](https://www.google.com/chrome/canary)), a `blockquote` is styled differently depending on the size of its container. 
+
+{{ embed.codepen(
+  id='YzpywrZ',
+  title='Container Queries with Block Quotes',
+  user='dvdherron',
+  height=600
+) }}
+
+### Define a containment context
+
+To achieve this, first a containment context must be defined. This lets the browser know which container to query against later. 
+
+```scss
+article,
+section {
+  contain: layout inline-size;
+}
+```
+
+The `contain` property is part of the existing [CSS Containment Module](https://drafts.csswg.org/css-contain/). The `layout` value turns on [layout contaiment](https://drafts.csswg.org/css-contain/#valdef-contain-layout) on the container, which ensures that "nothing outside can affect its internal layout, and vice versa." The `inline-size` value is a proposed change to the Containment Module that would let authors explicitly declare in which dimension containment should be applied. 
+
+### Apply @-rules at desired breakpoints
+Now that the containment context has been defined, the `@container` rule is used to tell the browser when and how styles should change. 
+
+```scss
+/* change styles according to container size */
+@container (min-width: 35em) {
+
+  .first {
+    --column: full;
+    margin-bottom: 0.5rem;
+  }
+
+  .title {
+    --title-background: forestgreen;
+  }
+
+  [data-quote] {
+    grid-gap: 1rem;
+    grid-template: "media quote" auto/ max-content 1fr;
+  }
+
+  .media {
+    align-items: center;
+  }
+
+  .source {
+    font-weight: bold;
+  }
+}
+
+@container (min-width: 40em) {
+
+  .spread {
+    --column: full;
+    display: grid;
+    grid-template-columns: minmax(50%, 1fr) minmax(min-content, 20em);
+    gap: 2rem;
+  }
+}
+
+@container (min-width: 60em) {
+
+  .title {
+    --title-background: goldenrod;
+  }
+
+  [data-quote] {
+    --quote-size: 1.5rem;
+    --quote-background: darkmagenta;
+    clip-path: polygon(
+      -0.04% 100.6%,
+      1.12% -0.52%,
+      95.65% 11.18%,
+      96.46% 90.05%
+    );
+    padding: 1.25em 3em;
+    max-width: unset;
+  }
+
+  .source {
+    font-size: 1.1rem;
+  }
+}
+```
+Any styles inside of one of these `@container` blocks will be applied when the container fulfills the condition of the query. 
+
+Just as with `@media` queries, authors can write as many `@container` queries as needed.
+
+## What's next?
+
+It's still very early in the proposal process, so a lot regarding how container queries work could change. 
+
+Block-size containment might also be possible, as well as querying against physical values like `width` and `height`:
+
+```css
+.block-container {
+  /* establishes a new containment context on the block axis */
+  contain: layout block-size;
+}
+```
+
+If containment on both inline and block axes is possible, queries might also be made against properties like `aspect-ratio` or `orientation`.
```

Comment:
Oh nice, I didn't even think about this. But we can actually contain both axis at once with `contain: size` (which is what you would need for these, I think). What we're not sure about is: can you contain _only the block axis_?

Reply:
Made this more clear and mentioned some of the other ways we might be able to query in the future.